packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart

```diff
@@ -79,94 +79,55 @@ class _MemoryChartPaneState extends State<MemoryChartPane>
 
   static int get _timestamp => DateTime.now().millisecondsSinceEpoch;
 
-  @override
-  void didChangeDependencies() {
-    super.didChangeDependencies();
-    if (!initController()) return;
-
-    // TODO(polinach): generalize three addAutoDisposeListener below.
-    addAutoDisposeListener(widget.chartController.event.tapLocation, () {
-      if (widget.chartController.event.tapLocation.value != null) {
-        if (_hoverOverlayEntry != null) {
-          _hideHover();
-        }
-        final tapLocation = widget.chartController.event.tapLocation.value;
-        if (tapLocation == null) return;
-        final tapDownDetails = tapLocation.tapDownDetails;
-        if (tapDownDetails == null) return;
-
-        final copied = TapLocation.copy(tapLocation);
-        widget.chartController.vm.tapLocation.value = copied;
-        widget.chartController.android.tapLocation.value = copied;
-
-        final allValues = ChartsValues(
-          controller,
-          tapLocation.index,
-          tapLocation.timestamp ?? _timestamp,
-        );
-
-        _showHover(
-          context,
-          allValues,
-          tapDownDetails.globalPosition,
-        );
-      }
-    });
-
-    addAutoDisposeListener(widget.chartController.vm.tapLocation, () {
-      if (widget.chartController.vm.tapLocation.value != null) {
-        if (_hoverOverlayEntry != null) {
-          _hideHover();
-        }
-        final tapLocation = widget.chartController.vm.tapLocation.value;
-        if (tapLocation == null) return;
-        final tapDownDetails = tapLocation.tapDownDetails;
-        if (tapDownDetails == null) return;
-
-        final copied = TapLocation.copy(tapLocation);
-        widget.chartController.event.tapLocation.value = copied;
-        widget.chartController.android.tapLocation.value = copied;
-
-        final allValues = ChartsValues(
-          controller,
-          tapLocation.index,
-          tapLocation.timestamp ?? _timestamp,
-        );
-
-        _showHover(
-          context,
-          allValues,
-          tapDownDetails.globalPosition,
-        );
-      }
-    });
+  void _addTapLocationListener(
+    ValueNotifier<TapLocation?> tapLocation,
+    List<ValueNotifier<TapLocation?>> allLocations,
+  ) {
+    addAutoDisposeListener(tapLocation, () {
+      final value = tapLocation.value;
+      if (value == null) return;
 
-    addAutoDisposeListener(widget.chartController.android.tapLocation, () {
       if (_hoverOverlayEntry != null) {
         _hideHover();
+        return;
```

Comment:
this added return is a logic change - do we want to return early after we hide the hover card? we didn't before adding this helper

Reply:
It seems we had bug. You needed to press 'Esc' to hide the popup window. Now it hides on click at the chart.